tools/check_code_style

```diff
@@ -31,5 +31,26 @@ fail() {
     exit 1
 }
 
-git grep -P 'check_screen.*00(?!.*nocheck:)' || exit 0
-fail "See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details"
+
+forbidden_check_var=0
+rules_violated=0
+files=$(git --no-pager diff --name-only HEAD $(git merge-base HEAD master))
+
+# rule 001
+for file in $files; do
+    if [[ -f $file   &&  $file == *.pm  && $file != *"Architectures.pm"*  &&  $file != *"Backends.pm"* ]]; then
+         grep -oHEn "check_var\('ARCH',.*\)|check_var\('BACKEND',.*\)" $file && ((forbidden_check_var+=1))
+    fi
+done
```

Comment:
I recommend to not iterate over individual files and apply custom filtering. You can still use `git grep` but filter out files to apply the grep to in there. See t/01_style.t for an example.

I just tried

```
git grep "check_var('\(ARCH\|BACKEND\)'," ':!lib/Utils/Architectures.pm' ':!lib/Utils/Backends.pm'
```

which seems to do what you need here.